### PR TITLE
PDI-40 PV2 387 Single level 1 aim progressing to single level 2 aim

### PR DESCRIPTION
**Pull request by CarlHA**

### CODE REVIEW

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/LearnerMutators/FM36Base.cs [hunk 1]
```diff
@@ -173,72 +178,110 @@ protected void ProcessMessageLearnerForLearnerRequestOriginatingFromTrainingReco
         protected void RemovePmrRecord(MessageLearner learner)
         {
             var deliveries = learner.LearningDelivery.ToList().Where(ld => ld.AimType == 1);
-            deliveries.ForEach(d=>d.AppFinRecord = d.AppFinRecord.ToList().Where(af => af.AFinType != LearnDelAppFinType.PMR.ToString()).ToArray());
+            deliveries.ForEach(d => d.AppFinRecord = d.AppFinRecord.ToList().Where(af => af.AFinType != LearnDelAppFinType.PMR.ToString()).ToArray());
         }
 
-        private void MutateAimForLearner(Aim aim, List<MessageLearnerLearningDelivery> learnerLearningDeliveries, int numberOfAimsForLearner)
+        private void MutateAimForLearner(List<Aim> aims, List<MessageLearnerLearningDelivery> learnerLearningDeliveries, int numberOfAimsForLearner)
```
Comment (kuzinator) [inline src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/LearnerMutators/FM36Base.cs]: `numberOfAimsForLearner` is a `Count` from the `aims` parameter, can this not be a separate parameter for clarity?

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/LearnerMutators/FM36Base.cs [hunk 2]
```diff
@@ -173,72 +178,110 @@ protected void ProcessMessageLearnerForLearnerRequestOriginatingFromTrainingReco
...
+            if (numberOfAimsForLearner == 1) // assume that this aim was created through the old style of Training records - in which case we need to setup the functionalskillsdelivery as well.
+            {
+                MutateAimType3ForLearnerFromTrainingRecord(learnerLearningDeliveries,
+                    learnerLearningDeliveries.Single(ld => ld.LearnAimRef == ProgrammeAim));
+            }
+        }
+
+        private void MutateOtherAimsForLearner(IEnumerable<Aim> otherAims, IEnumerable<MessageLearnerLearningDelivery> otherLearningDeliveries)
+        {
+            otherAims.OrderBy(x => x.AimSequenceNumber);
```
[the remark this answers is not in the document]
Comment (CarlHA) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/LearnerMutators/FM36Base.cs]: Updated

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/LearnerMutators/FM36Base.cs [hunk 3]
```diff
@@ -173,72 +178,110 @@ protected void ProcessMessageLearnerForLearnerRequestOriginatingFromTrainingReco
...
+            otherLearningDeliveries.OrderBy(x => x.AimSeqNumber);
```
[the remark this answers is not in the document]
Comment (CarlHA) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/LearnerMutators/FM36Base.cs]: updated

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/LearnerMutators/FM36Base.cs [hunk 4]
```diff
@@ -173,72 +178,110 @@ protected void ProcessMessageLearnerForLearnerRequestOriginatingFromTrainingReco
...
 
-            delivery.LearnAimRef = aim.AimReference;
+            MutateLearningDeliveryFamsForLearner(learningDelivery, aim);
 
-            delivery.LearnStartDate = aim.StartDate.ToDate();
-            delivery.LearnStartDateSpecified = true;
+            MutateProgrammeCodes(aim, learningDelivery);
+        }
+
+        private DateTime? MutateDates(Aim aim, MessageLearnerLearningDelivery learningDelivery)
```
[the remark this answers is not in the document]
Comment (CarlHA) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/LearnerMutators/FM36Base.cs]: Changed name

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/LearnerMutators/FM36Base.cs [hunk 5]
```diff
@@ -173,72 +178,110 @@ protected void ProcessMessageLearnerForLearnerRequestOriginatingFromTrainingReco
...
 
-            MutateCompletionStatusForLearner(delivery, (int)aim.CompletionStatus, actualEndDate);
 
-            MutateLearningDeliveryFamsForLearner(delivery, aim);
+            return actualEndDate;
+        }
 
-            delivery.ProgType = aim.ProgrammeType;
-            delivery.ProgTypeSpecified = true;
+        private void MutateProgrammeCodes(Aim aim, MessageLearnerLearningDelivery learningDelivery)
```
[the remark this answers is not in the document]
Comment (CarlHA) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/LearnerMutators/FM36Base.cs]: Changed to SetCourseCodes

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/LearnerMutators/FM36Base.cs (hunk 1 above)
Comment (CarlHA) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/LearnerMutators/FM36Base.cs]: Not sure I understand the comment.  numberOfAimsForLearner is a parameter in the method signature.

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/LearnerMutators/FM36Base.cs [hunk 6]
```diff
@@ -173,72 +178,110 @@ protected void ProcessMessageLearnerForLearnerRequestOriginatingFromTrainingReco
...
+                MutateAimType3ForLearnerFromTrainingRecord(learnerLearningDeliveries,
+                    learnerLearningDeliveries.Single(ld => ld.LearnAimRef == ProgrammeAim));
+            }
+        }
+
+        private void MutateOtherAimsForLearner(IEnumerable<Aim> otherAims, IEnumerable<MessageLearnerLearningDelivery> otherLearningDeliveries)
+        {
+            otherAims.OrderBy(x => x.AimSequenceNumber);
+            otherLearningDeliveries.OrderBy(x => x.AimSeqNumber);
+            for (var i = 0; i < otherAims.Count(); i++)
```
[the remark this answers is not in the document]
Comment (CarlHA) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/LearnerMutators/FM36Base.cs]: Updated

**Pull request merged**